src/backend/shortcuts/nonesteamgame/nonesteamgame.ts

```diff
@@ -259,8 +284,8 @@ async function addNonSteamGame(props: {
         otherGridAppID: generateShortAppId(newEntry.Exe, newEntry.AppName)
       },
       gameInfo: props.gameInfo,
-      bkgDataUrl: props.bkgDataUrl,
-      bigPicDataUrl: props.bigPicDataUrl
+      bkgDataUrl: bkgDataUrl!,
+      bigPicDataUrl: bigPicDataUrl!
```

Comment:
because of the `bkgDataUrl === undefined || bigPicDataUrl === undefined`, typescripts can't infer these should be strings at this point so I needed the `!`

Reply:
Not quite. This is caused by the return type of `window.webContents.executeJavaScript` being `Promise<any>`. So at this point, those two variables are `string | undefined` (not quite sure why it isn't `string | any`, I suspect some Type Narrowing is happening there)

Adding `as string` after the calls to `executeJavaScript` might make it more obvious what's happening.
It might be a good idea to make a helper function for this that has the return type `Promise<string>`, so you don't accidentally miss the `as string` if you ever need to call this function anywhere else

Reply:
that's right!